explorer/interpreter/type_checker.cpp

```diff
@@ -5908,11 +5908,15 @@ auto TypeChecker::DeclareAliasDeclaration(Nonnull<AliasDeclaration*> alias,
            << "invalid target for alias declaration";
   }
 
-  CARBON_ASSIGN_OR_RETURN(Nonnull<const Value*> target,
-                          InterpExp(&alias->target(), arena_, trace_stream_));
-
   alias->set_static_type(&alias->target().static_type());
-  alias->set_constant_value(target);
+  // constant_value not needed for namespace alias because these are resolved by
+  // NameResolver
```

Comment:
Do we need this special case? It would seem simpler to unconditionally set the constant value here if it's not doing any harm.

Reply:
otherwise `InterpExt(alias->target)` fails with the error like indicated in the removed fail_namespace_alias.test.  
>  could not find `namespace N;

This is because namespace declaration does not have constant value, just a static_value.

https://github.com/carbon-language/carbon-lang/blob/0dc80bac01acc21e15e72cf4ac060efa8a61bb59/explorer/interpreter/type_checker.cpp#L6072-L6076

I think this is because all namespace related expressions are replaced during name resolution.

> I think there are still cases where we'll look at this and assert / crash if it's not set.

This should because similar to namspace declaration, which also doesn't have a constant value.